src/webparts/todolist/components/ToDoList.tsx

```diff
@@ -1,129 +1,213 @@
-  Id: string;
-  Status: string;
-}
-
-export interface IListItemState {
-  items: ISPListItem[];
-  errorMessage: any;
-}
-
-class ItemList extends React.Component<IItemListProps, IListItemState> {
-  constructor(props: IItemListProps, state: IListItemState) {
-    super(props);
-
-    this.state = {
-      items: [],
-      errorMessage: null
-    };
-  }
-
-  private async _getListData(): Promise<ISPListItem[]> {
-    try {
-      const response = await this.props.spHttpClient.get(
-        `${this.props.webUrl}/_api/web/lists/getByTitle('To do list')/items?$select=Title,Status`,
-        SPHttpClient.configurations.v1
-      );
-
-      if (!response.ok) {
-        const responseText = await response.text();
-        throw new Error(responseText);
+import {
+  ISPListItem,
+  IItemListProps,
+  IToDoListProps
+} from "../../../interfaces";
+import { getSP } from "../../../pnpjsConfig";
+import { SPFI } from "@pnp/sp";
+import {
+  DetailsList,
+  DetailsListLayoutMode,
+  IColumn
+} from "@fluentui/react/lib/DetailsList";
+import { Text } from "@fluentui/react/lib/Text";
+import { IStackTokens, Stack, StackItem } from "@fluentui/react/lib/Stack";
+import { mergeStyles } from "@fluentui/react/lib/Styling";
+import { PrimaryButton } from "@fluentui/react/lib/Button";
+
+const ItemListStackTokens: IStackTokens = {
+  childrenGap: 15,
+  padding: 10
+};
+
+const ItemList: React.FC<IItemListProps> = (props: IItemListProps) => {
+  const [items, setItems] = React.useState<ISPListItem[]>([]);
+  const [errorMessage, setErrorMessage] = React.useState(null);
+  const columns: IColumn[] = [
+    {
+      key: "column1",
+      name: "Title",
+      fieldName: "Title",
+      minWidth: 100,
+      maxWidth: 200,
+      isResizable: true
+    },
+    {
+      key: "column2",
+      name: "Due Date",
+      minWidth: 100,
+      maxWidth: 200,
+      isResizable: true,
+      onRender: item => <Stack>{item.FieldValuesAsText.DueDate}</Stack>
+    },
+    {
+      key: "column3",
+      name: "Status",
+      fieldName: "Status",
+      minWidth: 100,
+      maxWidth: 200,
+      isResizable: true,
+      onRender: (item: ISPListItem) => {
+        switch (item.Status) {
+          case "Pending":
+            return (
+              <Stack
+                horizontalAlign="center"
+                verticalAlign="center"
+                className={mergeStyles({
+                  backgroundColor: "#ffe092",
+                  color: "#997825",
+                  borderRadius: "16px",
+                  height: "24px",
+                  whiteSpace: "nowrap",
+                  padding: "4px 8px",
+                  maxWidth: "max-content"
+                })}
+              >
+                {item.Status}
+              </Stack>
+            );
+
+          case "Completed":
+            return (
+              <Stack
+                horizontalAlign="center"
+                verticalAlign="center"
+                className={mergeStyles({
```

Comment:
I suggest doing it using css classes from scss file other than this way, or at least move all of these blocks of code to separate file.